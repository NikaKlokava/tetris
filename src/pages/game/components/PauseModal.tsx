import { Button } from "./Button";
import cl from "./styles.module.css";

type Props = {
  onPlay: () => void;
};

export const PauseModal = ({ onPlay }: Props) => {
  return (
    <div className={cl.modal_wrapper}>
      <div className={cl.modal_container}>
        <Button text="Resume" onPress={onPlay} />
      </div>
    </div>
  );
};
