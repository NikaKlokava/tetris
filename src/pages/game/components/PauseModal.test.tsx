import renderer from "react-test-renderer";
import { PauseModal } from "./PauseModal";

describe("Test the PauseModal component", () => {
  test("The PauseModal renders correctly", () => {
    const pauseModalSnap = renderer
      .create(
        <PauseModal
          onPlay={() => {
            console.log("resume game");
          }}
        />
      )
      .toJSON();
    expect(pauseModalSnap).toMatchSnapshot();
  });
});
