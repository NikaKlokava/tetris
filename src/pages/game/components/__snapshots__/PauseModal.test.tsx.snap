// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the GameContent component The GameContent renders correctly 1`] = `
<div
  className="modal_wrapper"
>
  <div
    className="modal_container"
  >
    <button
      className="game_btn"
      onClick={[Function]}
    >
      Resume
    </button>
  </div>
</div>
`;

exports[`Test the PauseModal component The PauseModal renders correctly 1`] = `
<div
  className="modal_wrapper"
>
  <div
    className="modal_container"
  >
    <button
      className="game_btn"
      onClick={[Function]}
    >
      Resume
    </button>
  </div>
</div>
`;
